### PR TITLE
Fix confirmation location changed on PutAway

### PR DESCRIPTION
**Pull request by TDu**
Also added the same change on Pallet Transfer

### CODE REVIEW

File: shopfloor/actions/message.py [hunk 1]
```diff
@@ -92,6 +92,13 @@ def dest_location_not_allowed(self):
     def need_confirmation(self):
         return {"message_type": "warning", "message": _("Are you sure?")}
 
+    def confirm_location_changed(self, from_location, to_location):
+        return {
+            "message_type": "warning",
+            "message": _("Confirm location change from %s to %s ?")
```
Comment (guewen) [inline shopfloor/actions/message.py]:
```suggestion
            "message": _("Confirm location change from %s to %s?")
```

File: shopfloor/actions/message.py [hunk 2]
```diff
@@ -92,6 +92,13 @@ def dest_location_not_allowed(self):
     def need_confirmation(self):
         return {"message_type": "warning", "message": _("Are you sure?")}
 
+    def confirm_location_changed(self, from_location, to_location):
```
Comment (guewen) [inline shopfloor/actions/message.py:95]: I suggest passing a browse record here, so people who want to customize the message can use any value from a location.

File: shopfloor/services/cluster_picking.py [hunk 3]
```diff
@@ -112,7 +112,8 @@ def list_batch(self):
     def _select_a_picking_batch(self, batches):
         # look for in progress + assigned to self first
         candidates = batches.filtered(
-            lambda batch: batch.state == "in_progress" and batch.user_id == self.env.user
+            lambda batch: batch.state == "in_progress"
+            and batch.user_id == self.env.user
```
Comment (guewen) [inline shopfloor/services/cluster_picking.py]: Is it what black did with pre-commit? It doesn't do this on my side

File: shopfloor/services/single_pack_putaway.py [hunk 4]
```diff
@@ -191,10 +191,14 @@ def _response_for_forbidden_location(self, move_line, pack):
             data=self._data_after_package_scanned(move_line, pack),
         )
 
-    def _response_for_location_need_confirm(self):
+    def _response_for_location_need_confirm(self, move_line, pack, to_location):
```
Comment (guewen) [inline shopfloor/services/single_pack_putaway.py:194]: `to_location` as recordset?

**Comment by guewen** [comment]
do you use pre-commit or do you run `black` manually? (different versions of black can give different results, but `pre-commit` pins a version so you should use it if not yet)

File: shopfloor/services/cluster_picking.py (hunk 3 above)
Comment (TDu) [reply shopfloor/services/cluster_picking.py]: Travis was complaining about it (or was it only a warning ?) I may have forced it with black to get the unit tests working. Will revert back to see what happens...

**Comment by TDu** [comment]
@guewen Yes I am using pre-commit.

File: shopfloor/actions/message.py (hunk 2 above)
Comment (TDu) [reply shopfloor/actions/message.py:95]: Yes changed

File: shopfloor/actions/message.py (hunk 1 above)
Comment (TDu) [reply shopfloor/actions/message.py]: Fixed

File: shopfloor/services/single_pack_putaway.py (hunk 4 above)
Comment (TDu) [reply shopfloor/services/single_pack_putaway.py:194]: Changed

**Comment by simahawk** [comment]
rebased, squashed and merged manually

**Pull request closed without merging**